ghcide/src/Development/IDE/Plugin/Completions.hs

```diff
@@ -106,6 +109,44 @@ dropListFromImportDecl iDecl = let
     f x = x
     in f <$> iDecl
 
+resolveCompletion :: IdeState -> PluginId -> CompletionItem -> LSP.LspM Config (Either ResponseError CompletionItem)
+resolveCompletion ide _ comp@CompletionItem{_detail,_documentation,_xdata}
+  | Just resolveData <- _xdata
+  , Success (uri, NameDetails mod occ) <- fromJSON resolveData
+  , Just file <- uriToNormalizedFilePath $ toNormalizedUri uri = do
+    mdkm <- liftIO $ runAction "Completion resolve" ide $ use GetDocMap file
```

Comment:
Why is this in its own action, whereas below we have multiple `use`s in the same action? Worth a comment.